src/Raven.Client/Documents/Operations/AttachmentResult.cs

```diff
@@ -1,12 +1,40 @@
-﻿namespace Raven.Client.Documents.Operations
+﻿using System;
+using System.IO;
+using System.Net.Http;
+
+namespace Raven.Client.Documents.Operations
 {
-    public class AttachmentResult
+    public class AttachmentResultWithStream : AttachmentResult, IDisposable
```

Comment:
Don't think it is a good idea, we can just return a `Stream` directly from the method, don't need this.
We can also have a `out` param for the `AttachmentDetails`